tests/scripts/all.sh

```diff
@@ -115,12 +115,19 @@ set -e -o pipefail -u
 # Enable ksh/bash extended file matching patterns
 shopt -s extglob
 
+# For project detection
+PROJECT_NAME_FILE='./scripts/project_name.txt'
+if read -r PROJECT_NAME < "$PROJECT_NAME_FILE"; then :; else
+    echo "$PROJECT_NAME_FILE does not exist... Exiting..." >&2
+    exit 1
+fi
```

Comment:
Please don't add more behavior at the top level. All of this new code should be inside the `pre_check_environment` function.